src/util/safelywritablefile.cpp

```diff
@@ -263,6 +243,75 @@ bool SafelyWritableFile::commit() {
     return true;
 }
 
+bool SafelyWritableFile::useTempFileWithPrefix() {
+    if (!QFileInfo(m_origFileName).isWritable()) {
```

Comment:
This check is common for all and should be moved to the constructor. The move to m_origFileName should be happen afterwards.